crates/pallet-pot/src/lib.rs

```diff
@@ -190,14 +200,39 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
     }
 }
 
-impl<T: Config<I>, I: 'static> OnUnbalanced<NegativeImbalanceOf<T, I>> for Pallet<T, I> {
+/// Handle unbalanced funds by depositing them into this pot.
+///
+/// Implementation for [`Currency`].
+pub struct DepositUnbalancedCurrency<T, I>(PhantomData<(T, I)>);
+
+impl<T: Config<I>, I: 'static> OnUnbalanced<NegativeImbalanceOf<T, I>>
+    for DepositUnbalancedCurrency<T, I>
+{
     fn on_nonzero_unbalanced(amount: NegativeImbalanceOf<T, I>) {
         let numeric_amount = amount.peek();
 
         // Must resolve into existing but better to be safe.
-        T::Currency::resolve_creating(&Self::account_id(), amount);
+        T::Currency::resolve_creating(&Pallet::<T, I>::account_id(), amount);
+
+        Pallet::<T, I>::deposit_event(Event::Deposit {
+            value: numeric_amount,
+        });
+    }
+}
+
+/// Handle unbalanced funds by depositing them into this pot.
+///
+/// Implementation for [`Fungible`].
```

Comment:
Fix the clippy offense

Reply:
https://github.com/humanode-network/humanode/pull/861/commits/12704682582c189efb32875110c7962981892c73